### ISSUE TITLE
markdown special symbols in commit description should be escaped

### ISSUE THREAD

**Issue by ChALkeR** [nodejs/changelog-maker#47]
The commit messages are initially in text, so rendering them as Markdown can cause errors.

To reproduce, in `node`, do:
```console
$ changelog-maker --start-ref=9eddaebb --end-ref=9eddaebb
* [[`9eddaebb79`](https://github.com/nodejs/node/commit/9eddaebb79)] - **assert**: remove unnecessary use of __proto__ (Nathan Rajlich) 
```

If a changelog includes `__proto__`, we don't want it to render as «__proto__», we want to render it as «`__proto__`» or «\_\_proto\_\_». Without knowing more if that is code block or not, it would make sense to just escape some symbols like <code>`_*~\\</code>, prefixing those with <code>\\</code>.

Refs: [nodejs/node#a58b48bc3bc](https://github.com/nodejs/node/commit/a58b48bc3bcf43f7090d4cc914606af68fe55815#diff-869344da7e479f3aa0fbe085f3ee688dL715), https://github.com/nodejs/node/pull/7637, https://github.com/nodejs/node/pull/12756#discussion_r114107415.

**Comment by wooorm** [comment]
Hi folks! 👋

Shameless plug, but maybe [remark](https://github.com/wooorm/remark) can help here, as it automatically deals with escapes and stuff! [Here’s an example of how it could look](https://github.com/wooorm/remark-lint/blob/c31c0e1023e0b309a427b5e8a20e098825f26f64/script/build-presets.js#L58-L102), where a readme is generated from a `package.json` in a monorepo.

Let me know if that’s of interest!

**Comment by MylesBorins** [comment]
BUMP to this

This actually broke CI today as it broke the markdown linter

https://github.com/nodejs/node/pull/18767

**Comment by rvagg** [comment]
fixed in #52, pls :+1:
